### PR TITLE
refactor(mac): remove unused context calls 🌱

### PR DESCRIPTION
**Pull request by mcdurdin**
Relates to #10365.

Ensures that the only uses of the Core context are for unit tests. Renames accordingly to clarify usage.

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by mcdurdin** [comment]
Well, not compiling locally means a lot of fiddly little commits sorry. Could squash those but it's not all that important.

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/CoreWrapperTests.m [hunk 1]
```diff
@@ -67,8 +67,8 @@ - (void)testgetContextAsString_ContextContainsEmojis_ReturnsSameContext  {
   NSString *kmxPath = [CoreTestStaticHelperMethods getKmxFilePathTestMacEngine];
   CoreWrapper *core = [[CoreWrapper alloc] initWithHelper: [CoreTestStaticHelperMethods helper] kmxFilePath:kmxPath];
   [core setContext:@"🤔?👍🏻✅"];
-  NSString *finalContext = core.context;
-  XCTAssert([finalContext isEqualToString:@"🤔?👍🏻✅"], @"Expected '🤔?👍🏻✅' in context buffer");
+  NSString *finalContext = core.contextDebug;
+  XCTAssert([finalContext isEqualToString:@"|🤔?👍🏻✅| (len: 5) [ U+1f914 U+003f U+1f44d U+1f3fb U+2705 ]"], @"Expected '🤔?👍🏻✅' in context buffer");
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/CoreWrapperTests.m:71]:
```suggestion
  // Note: relying on km_core_state_context_debug output format is just barely
  // acceptable for a unit test
  XCTAssert([finalContext isEqualToString:@"|🤔?👍🏻✅| (len: 5) [ U+1f914 U+003f U+1f44d U+1f3fb U+2705 ]"], @"Expected '🤔?👍🏻✅' in context buffer");
```

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m [hunk 2]
```diff
@@ -554,8 +562,8 @@ - (void)testCoreProcessEvent_backspaceElNuerWithContext_EmptiesContextReturnsDel
     NSLog(@"output: %@", output);
     XCTAssert(output.codePointsToDeleteBeforeInsert == 1, @"Expected output to delete one code point");
     XCTAssert(!output.hasTextToInsert, @"expected to insert nothing");
-    NSString *context = engine.getCoreContext;
-    XCTAssert([context isEqualToString:@""], @"Context should be empty.");
+    NSString *context = engine.getCoreContextDebug;
+    XCTAssert([context isEqualToString:@"|| (len: 0) [ ]"], @"Context should be empty.");
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m:566]:
```suggestion
    // Note: relying on km_core_state_context_debug output format is just barely
    // acceptable for a unit test
    XCTAssert([context isEqualToString:@"|| (len: 0) [ ]"], @"Context should be empty.");
```

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m [hunk 3]
```diff
@@ -564,8 +572,8 @@ - (void)testCoreProcessEvent_eventReturnWithElNuerKmx_EmitWithContextEmpty {
     NSEvent *event = [NSEvent keyEventWithType:NSEventTypeKeyDown location:NSMakePoint(0, 0) modifierFlags:0 timestamp:0 windowNumber:0 context:nil characters:@"\n" charactersIgnoringModifiers:@"\n" isARepeat:NO keyCode:kVK_Return];
     CoreKeyOutput *output = [engine processEvent:event];
     XCTAssert(output.emitKeystroke, @"Expected emitKeystroke==YES");
-    NSString *context = engine.getCoreContext;
-    XCTAssert([context isEqualToString:@""], @"Context should be cleared.");
+    NSString *context = engine.getCoreContextDebug;
+    XCTAssert([context isEqualToString:@"|| (len: 0) [ ]"], @"Context should be cleared.");
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m:576]:
```suggestion
    // Note: relying on km_core_state_context_debug output format is just barely
    // acceptable for a unit test
    XCTAssert([context isEqualToString:@"|| (len: 0) [ ]"], @"Context should be cleared.");
```

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m [hunk 4]
```diff
@@ -574,8 +582,8 @@ - (void)testCoreProcessEvent_eventTabWithElNuerKmx_EmitWithContextEmpty {
     NSEvent *event = [NSEvent keyEventWithType:NSEventTypeKeyDown location:NSMakePoint(0, 0) modifierFlags:0 timestamp:0 windowNumber:0 context:nil characters:@"\t" charactersIgnoringModifiers:@"\t" isARepeat:NO keyCode:kVK_Tab];
     CoreKeyOutput *output = [engine processEvent:event];
     XCTAssert(output.emitKeystroke, @"Expected emitKeystroke==YES");
-    NSString *context = engine.getCoreContext;
-    XCTAssert([context isEqualToString:@""], @"Context should be cleared.");
+    NSString *context = engine.getCoreContextDebug;
+    XCTAssert([context isEqualToString:@"|| (len: 0) [ ]"], @"Context should be cleared.");
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m:586]:
```suggestion
    // Note: relying on km_core_state_context_debug output format is just barely
    // acceptable for a unit test
    XCTAssert([context isEqualToString:@"|| (len: 0) [ ]"], @"Context should be cleared.");
```

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m [hunk 5]
```diff
@@ -585,8 +593,8 @@ - (void)testCoreProcessEvent_eventSingleQuoteWithElNuerKmx_ReturnsDiacritic {
     NSEvent *event = [NSEvent keyEventWithType:NSEventTypeKeyDown location:NSMakePoint(0, 0) modifierFlags:0 timestamp:0 windowNumber:0 context:nil characters:@"'" charactersIgnoringModifiers:@"'" isARepeat:NO keyCode:kVK_ANSI_Quote];
     CoreKeyOutput *output = [engine processEvent:event];
     XCTAssert(output.hasTextToInsert, @"returns text to insert");
-    context = engine.getCoreContext;
-    XCTAssert([context isEqualToString:@"\u025B\u0308"], @"Context updated with diacritic.");
+    context = engine.getCoreContextDebug;
+    XCTAssert([context isEqualToString:@"|\u025B\u0308| (len: 2) [ U+025b U+0308 ]"], @"Context updated with diacritic.");
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m:597]:
```suggestion
    // Note: relying on km_core_state_context_debug output format is just barely
    // acceptable for a unit test
    XCTAssert([context isEqualToString:@"|\u025B\u0308| (len: 2) [ U+025b U+0308 ]"], @"Context updated with diacritic.");
```

Comment (darcywong00) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m:597]: Is `getCoreContextDebug` in the unit tests going to be long-term?

Could replace `\u025B\u0308` with `ɛ̈`

**Review by darcywong00 — APPROVED** [review]
lgtm

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m (hunk 5 above)
Comment (mcdurdin) [reply mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m:597]: > Is `getCoreContextDebug` in the unit tests going to be long-term?

Well for the foreseeable future, yes. Why?

> Could replace `\u025B\u0308` with `ɛ̈`

I think it's clearer to be able to see the combining diacritic is a separate codepoint, this way.

Comment (darcywong00) [reply mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m:605]: Gotcha. I just found it nice in CoreWrapperTests.m to see the emoji characters and corresponding codepoints.

**Review by sgschantz — APPROVED** [review]
This looks good. I think the new names using 'ContextDebug' make it clear about how to use the context from core. However, using the values for unit tests seems to exceed the limitation imposed by the name 'Debug' -- if a unit test can rely on the expected context, then we should be able to do the same with other platform code.

The comments about the unit tests being 'barely acceptable' reinforce this idea. I read that as it being actually unacceptable, but we'll let it go this time.  I don't have a better idea for the naming, and it still makes me feel better to have unit tests that check the context. Hopefully others that read this code later won't be too confused by the contradiction.

**Comment by mcdurdin** [comment]
@sgschantz yes. I think this arises because the surface we are testing is actually Core and not Engine. In effect we are relying on a side-effect from an internal library to ensure that things are working, and we should probably be mocking Core away for our tests. But what we have now does work and it's also not necessarily worth the effort to go further (as in, we can better spend the effort elsewhere)

**Pull request merged**